### ISSUE TITLE
加上正则化向或者dropout或者bn能不能防止过拟合呢？

### ISSUE THREAD

**Issue by yangbisheng2009** [alvenchen/nsfw-pytorch#3]
我在训练的过程中，现在跑了大约30个epoch
发现训练集的准确率总是高于验证集10个点，感觉像是有点过拟合，
我现在准备改一下，添加一些防止过拟合的参数，希望楼主能给点建议

**Comment by alvenchen** [comment]
现在准确度多少了啊？可以增加 data augmentation，dropout， 减少batch_size试下
其实主要还是脏数据太多了

**Comment by yangbisheng2009** [comment]
训练集： 97
验证集：89
我现在batch_size 设置的256，减小batch_size是不是训练的更慢了？

2225 training * Prec@1 97.354
2226 /data/user/yangfg/software/Anaconda2/envs/py2/lib/python2.7/site-packages/PIL/TiffImagePlugin.py:785: UserWarning: Corrupt EXIF data.  Expecting to read 2 bytes but only got 0.
2227   warnings.warn(str(msg))
2228 /data/user/yangfg/software/Anaconda2/envs/py2/lib/python2.7/site-packages/PIL/Image.py:2618: DecompressionBombWarning: Image size (98130452 pixels) exceeds limit of 89478485 pixels, could
     be decompression bomb DOS attack.
2229   DecompressionBombWarning)
2230 evaluate * Prec@1 89.561
2231 Epoch [22], Iter [1/985]  47.7283068326 samples/sec, Losses: 0.130618542433

**Comment by alvenchen** [comment]
等下，你这是准确率啊，那不是过拟合啊
继续训练呗，应该还没收敛吧

**Comment by yangbisheng2009** [comment]
嗯嗯，感觉效果还不错
对了关于数据增强，你为啥要自己写呢？我用了torchvision自带的一些数据增强方法，貌似没你的效果好，你再训练的过程中，有尝试哪个数据增强对效果影响最大呢？

**Comment by alvenchen** [comment]
都可以的，只不过熟悉了pillow或者opencv就习惯用他们写了，而且改成tensorflow也容易
Gaussian，color，Blur等等，感觉都挺有用的啊，不仅相当于增加了数据集，还相当于regularization了，也就减少了过拟合。
